### PR TITLE
Fixes #1143 select individual when exporting container from spatial structure

### LINKED ISSUE #1143: Select individual when exporting container from Spatial Structure
The list of parameters has to be completed by the selected individual, and all parameters have to be exported.
Look through the individual for paths that match the container

Implement as an additional context menu entry and keep the existing 'Save as PKML...' when user wants to just export the container

### PR DESCRIPTION
**Pull request by rwmcintosh**
You open the context menu and select 'Export With Individual'.  Any suggestions for this context menu are appreciated
image

Can't close without specifying both individual and file path. The file path button opens the standard file browser.
image

Ok, saves the container PKML with additional parameters from the individual where they match the container path
image

### CODE REVIEW

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs [hunk 1]
```diff
@@ -40,45 +63,98 @@ protected override IEnumerable<string> GetUnallowedNames(IContainer container, I
...
+      private void addIndividualParametersToContainer(IndividualBuildingBlock individual, IContainer container, ObjectPath containerPath)
+      {
+         
+         individual.Where(individualParameter => individualParameter.ContainerPath.Equals(containerPath)).Each(x => addIndividualParameterToContainer(x, container));
+      }
 
+      private void addIndividualParameterToContainer(IndividualParameter individualParameter, IContainer container)
+      {
+         var parameter = _individualParameterToParameterMapper.MapFrom(individualParameter);
+         if (container.ContainsName(parameter.Name))
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:143]: I'm not 100% sure on this. The scenario is that the parameter is defined in both individual and container. For now, I'm replacing the parameter from the spatial structure with a new one from the individual.

Mostly, this did not happen, but I ran into a case when exporting the 'Organism' where 'MeanHeight' is declared in both.

**Comment by rwmcintosh** [comment]
@PavelBal @georgeDaskalakis Do you have any suggestions for a concise context menu entry for this feature?

**Comment by rwmcintosh** [comment]
Converting to draft. I'll see if it's possible to handle the renaming during import since https://github.com/Open-Systems-Pharmacology/MoBi/issues/1155 was implemented already

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs (hunk 1 above)
Comment (Yuri05) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:143]: Looks good for me so far.

**Review by msevestre — CHANGES_REQUESTED** [review]
Nothing wrong but I think this can be improved.
Resharper should be run however => request change

File: src/MoBi.UI/Views/SelectFolderAndIndividualFromProjectView.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      }
+
+      public void AttachPresenter(ISelectFolderAndIndividualFromProjectPresenter presenter)
+      {
+         _presenter = presenter;
+      }
+
+      public void BindTo(IndividualAndFilePathDTO dto)
+      {
+         descriptionLabel.Text = ExportContainerDescription(dto.ContainerPath.PathAsString);
```
[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/MoBi.UI/Views/SelectFolderAndIndividualFromProjectView.cs:50]: yes, it works.

File: src/MoBi.Presentation/Presenter/SelectFolderAndIndividualFromProjectPresenter.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+using OSPSuite.Presentation.Presenters;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface ISelectFolderAndIndividualFromProjectPresenter : IDisposablePresenter
+   {
+      /// <summary>
+      ///    Opens a dialog where the user will select an IndividualBuildingBlock from the project and a file path to export
+      /// </summary>
+      (string, IndividualBuildingBlock) GetPathAndIndividualForExport(IContainer container);
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/SelectFolderAndIndividualFromProjectPresenter.cs:18]: I think you can name the return values also but I am not sure (string exportPath, xxx)

**Pull request merged**